app/server.py

```diff
@@ -31,37 +32,36 @@ def bad_request(err):
 
 @app.route('/api/v1/search/<search_engine>', methods=['GET'])
 def search(search_engine):
+    search_engine = search_engine.strip().lower()
     try:
         num = request.args.get('num') or 10
         count = int(num)
-        qformat = request.args.get('format') or 'json'
+        qformat = request.args.get('format', 'json')
         if qformat not in ('json', 'xml'):
             abort(400, 'Not Found - undefined format')
 
-        engine = search_engine
-        if engine not in ('google', 'bing', 'duckduckgo', 'yahoo', 'ask',
-                          'yandex', 'ubaidu'):
-            err = [404, 'Incorrect search engine', qformat]
-            return bad_request(err)
-
         query = request.args.get('query')
         if not query:
             err = [400, 'Not Found - missing query', qformat]
             return bad_request(err)
 
-        result = feedgen(query, engine[0], count)
+        try:
+            result = feedgen(query, search_engine, count)
+        except KeyError:  # Unsupported search_engine
```

Comment:
This can throw any kind of error, may be runtime and not always due to unsupported search engine

Reply:
Agreed.  I am only catching the issue that I know how to deal with.  All other errors will continue to function as before.

Reply:
Agree. Look into this.

Reply:
Again.  My intention is not to create any new side effects with this change.  My goal is only to consolidate the information about search engines into a single dict in a single file.  Catching other errors should be handled in another PR dedicated to that purpose.